tests/test_all_fields.py

```diff
@@ -0,0 +1,187 @@
+        for stream, records in expected_records.items():
+            print("Ensuring expected data for {} has values formatted correctly.".format(stream))
+            self.modify_expected_datatypes(records)
+
+        # Instantiate connection with default start/end dates
+        conn_id = connections.ensure_connection(self)
+
+        # run in check mode
+        check_job_name = runner.run_check_mode(self, conn_id)
+
+        # verify check exit codes
+        exit_status = menagerie.get_exit_status(conn_id, check_job_name)
+        menagerie.verify_check_exit_status(self, exit_status, check_job_name)
+
+        found_catalogs = menagerie.get_catalogs(conn_id)
+        self.assertGreater(len(found_catalogs), 0, msg="unable to locate schemas for connection {}".format(conn_id))
+
+        found_catalog_names = set(map(lambda c: c['tap_stream_id'], found_catalogs))
+        diff = self.expected_check_streams().symmetric_difference( found_catalog_names )
+        self.assertEqual(len(diff), 0, msg="discovered schemas do not match: {}".format(diff))
+        print("discovered schemas are OK")
+
+        # Select all available fields from all streams
+        self.select_all_streams_and_fields(conn_id=conn_id, catalogs=found_catalogs, select_all_fields=True)
+
+        catalogs = menagerie.get_catalogs(conn_id)
+
+        # Ensure our selection worked
+        for cat in found_catalogs:
+            catalog_entry = menagerie.get_annotated_schema(conn_id, cat['stream_id'])
+            # Verify all streams are selected
+            selected = catalog_entry.get('annotated-schema').get('selected')
+            print("Validating selection on {}: {}".format(cat['stream_name'], selected))
+            self.assertTrue(selected, msg="Stream not selected.")
+
+            # Verify all fields within each stream are selected
+            for field, field_props in catalog_entry.get('annotated-schema').get('properties').items():
+                field_selected = field_props.get('selected')
+                print("\tValidating selection on {}.{}: {}".format(cat['stream_name'], field, field_selected))
+                self.assertTrue(field_selected, msg="Field not selected.")
+
+        #clear state
+        menagerie.set_state(conn_id, {})
+
+        # run sync
+        sync_job_name = runner.run_sync_mode(self, conn_id)
+
+        # Verify tap exit codes
+        exit_status = menagerie.get_exit_status(conn_id, sync_job_name)
+        menagerie.verify_sync_exit_status(self, exit_status, sync_job_name)
+
+        # read target output
+        first_record_count_by_stream = runner.examine_target_output_file(self, conn_id,
+                                                                         self.expected_streams(),
+                                                                         self.expected_primary_keys())
+        replicated_row_count =  reduce(lambda accum,c : accum + c, first_record_count_by_stream.values())
```

Comment:
Instead of reduce can you just use sum here?

Reply:
Good point. Not sure why reduce was used, I think sum would be more clear.